# Probability
The most popular way to express probability is as a percentage, as in “There is a 70% chance my flight will be late.” We will call this probability P X , where X is the event of interest. As you work with probabilities, though, you will more likely see it expressed as a decimal (in this case .70), which must be between 0.0 and 1.0:
P(X) =.70
Sometimes people use the terms probability and statistics interchangeably, and while it is understandable to conflate the two disciplines, they do have distinctions. Proba‐ bility is purely theoretical of how likely an event is to happen and does not require data. Statistics, on the other hand, cannot exist without data and uses it to discover probability and provides tools to describe data.

# Joint probability
Let’s say you have a fair coin and a fair six-sided die. You want to find the probability of flipping a heads and rolling a 6 on the coin and die, respectively. These are two separate probabilities of two separate events, but we want to find the probability that both events will occur together. This is known as a joint probability:
P(A AND B) = P(A) × P(B)

# Union Probabilities
When we deal with OR operations with probabilities, this is known as a union probability.
Let’s start with mutually exclusive events, which are events that cannot occur simulta‐ neously. For example, if I roll one die I cannot simultaneously get a 4 and a 6. I can only get one outcome. Getting the union probability for these cases is easy. I simply add them together. If I want to find the probability of getting a 4 or 6 on a die roll, it is going to be 2/6 = 1/3:
P(4) = 1/6
P(6) = 1/6
P(4 OR 6) =1/6+1/6=1/3

# Sum rule of probability
Sum rule of probability ensures every joint event is counted only once:
P(A OR B) = P(A) + P(B) − P(A AND B)
P(A OR B) = P(A) + P(B) − P(A) * P(B)

# Conditional probability
A probability topic that easily confuses people is the concept of conditional probabil‐ ity, which is the probability of an event A occurring given event B has occurred. It is typically expressed as P(A GIVEN B) or P(A|B).

# Bayes' Theorem
P(A|B) = P(B|A) * P(A) / P(B).
Using Bayes’ Theorem in Python:

In [1]:
p_coffee_drinker = .65
p_cancer = 0.005
p_coffee_drinker_given_cancer = 0.85

In [3]:
p_cancer_given_coffee_drinker = p_coffee_drinker_given_cancer * p_cancer / p_coffee_drinker

In [4]:
print(p_cancer_given_coffee_drinker)

0.006538461538461539


So the probability someone has cancer given they are a coffee drinker is only 0.65%! This number is very different from the probability someone is a coffee drinker given they have cancer, which is 85%.

#Binomial distribution
Binomial distribution measures how likely k successes can happen out of n trials given p probability.
Using SciPy for the binomial distribution:

In [5]:
from scipy.stats import binom

In [6]:
n = 10
p = 0.9

In [7]:
for k in range(n + 1):
  probability = binom.pmf(k, n, p)
  print("{0} - {1}".format(k, probability))

0 - 9.999999999999978e-11
1 - 8.999999999999978e-09
2 - 3.6449999999999943e-07
3 - 8.747999999999991e-06
4 - 0.00013778099999999974
5 - 0.0014880347999999982
6 - 0.011160260999999996
7 - 0.05739562799999997
8 - 0.1937102444999998
9 - 0.38742048899999976
10 - 0.3486784401000001


# Beta Distribution
The beta distribution allows us to see the likelihood of different underlying probabilities for an event to occur given alpha successes and beta failures.
The beta distribution is a type of probability distribution, which means the area under the entire curve is 1.0, or 100%. To find a probability, we need to find the area within a range.
Beta distribution using SciPy:

In [8]:
from scipy.stats import beta

In [9]:
a = 8
b = 2

In [12]:
p = 1.0 - beta.cdf(.90, a, b)

In [13]:
print(p)

0.22515902199999982


This means that out of 8/10 successful engine tests, there is only a 22.5% chance the underlying success rate is 90% or greater. But there is about a 77.5% chance it is less than 90%. The odds are not in our favor here that our tests were successful, but we could gamble on that 22.5% chance with more tests if we are feeling lucky.